iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/TimeCalculator.java

```diff
@@ -4,11 +4,10 @@
 import java.time.LocalDateTime;
 
 public class TimeCalculator {
-	public int calculateRelativeTime(SwitchingMoment switchingMoment, LocalDateTime scheduledTime) {
-		LocalDateTime triggerTime = switchingMoment.getTriggerTime();
-		Duration relativeTime = Duration.between(scheduledTime, triggerTime);
+    public static int calculateSecondsUntil(LocalDateTime from, LocalDateTime until) {
+        Duration relativeTime = Duration.between(until, from);
```

Comment:
This calculates the time from "until" to "from". It means you've switched those two arguments.
In the call to this method, the currentTime is given as second argument. currentTime is the "fromTime", so it should be the first argument.

Reply:
Done, along with migrating the changes to the method calls.